docs/markdown/Reference-manual.md

```diff
@@ -1507,11 +1511,19 @@ the following methods:
   an array containing only the arguments supported by the compiler,
   as if `has_argument` were called on them individually.
 
+- `get_supported_link_arguments(list_of_string)` *(added 0.46.0)* returns
+  an array containing only the arguments supported by the linker,
+  as if `has_link_argument` were called on them individually.
+
 - `has_argument(argument_name)` returns true if the compiler accepts
   the specified command line argument, that is, can compile code
-  without erroring out or printing a warning about an unknown flag,
-  you can specify external dependencies to use with `dependencies`
-  keyword argument.
+  without erroring out or printing a warning about an unknown flag.
+
+- `has_link_argument(argument_name)` *(added 0.46.0)* returns true if the linker
+  accepts the specified command line argument, that is, can compile and link
+  code without erroring out or printing a warning about an unknown flag. Note
+  that linker arguments will be converted to compiler arguments by prepending
+  '-Wl,' if needed, or by adding '/link' argument for VisualStudio.
```

Comment:
How does this work for linker arguments that take arguments? Suppose you want to test `-z global`. The compiler version would be `-Wl,z,global but for the linker it would be `-z global`. What if the user tries `has_link_argument('-z', 'global')? How would we know when to join them with commas? Are there cases where combining everything with commas would be the wrong thing to do?

Would it make sense to have users write the linker flags to check in the `-Wl` format instead (but not for VS)? Based on existing build definition files I looked at, this is what they seem to already be doing.